File: src/main/scala/com/dadrox/sbt/junit/JunitFramework.scala
package com.dadrox.sbt.junit

import org.scalatools.testing.Logger
import org.scalatools.testing.Runner
import org.scalatools.testing
import scala.collection.mutable.ListBuffer
import java.util.concurrent.atomic.AtomicInteger

// TODO?
// add:
// test-only filtering
//

class JunitFramework extends testing.Framework {
    override val name = "Junit"
    override val tests = Array[testing.Fingerprint](Junit3Fingerprint, Junit4Fingerprint)

    override def testRunner(classLoader: ClassLoader, loggers: Array[Logger]) = new JunitRunner(classLoader, loggers)
}

object Junit3Fingerprint extends testing.SubclassFingerprint {
    override val superClassName = "junit.framework.TestCase"
    override val isModule = false
}

object Junit4Fingerprint extends testing.AnnotatedFingerprint {
    override val annotationName = "org.junit.Test"
    override val isModule = false
}

File: src/main/java/com/dadrox/sbt/junit/JunitFramework.java
package com.dadrox.sbt.junit;

import org.scalatools.testing.AnnotatedFingerprint;
import org.scalatools.testing.Fingerprint;
import org.scalatools.testing.Logger;
import org.scalatools.testing.Runner;
import org.scalatools.testing.Framework;
import org.scalatools.testing.SubclassFingerprint;

public class JunitFramework implements Framework {
    private static SubclassFingerprint Junit3Fingerprint = new SubclassFingerprint() {
        @Override
        public boolean isModule() {
            return false;
        }

        @Override
        public String superClassName() {
            return "junit.framework.TestCase";
        }
    };

    private static AnnotatedFingerprint Junit4Fingerprint = new AnnotatedFingerprint() {
        @Override
        public String annotationName() {
            return "org.junit.Test";
        }

        @Override
        public boolean isModule() {
            return false;
        }
    };

    @Override
    public String name() {
        System.out.println("Someone get name!");

        return "Junit";
    }

    @Override
    public Runner testRunner(ClassLoader classLoader, Logger[] loggers) {
        return new JunitRunner(classLoader, loggers);
    }

    @Override
    public Fingerprint[] tests() {
        return new Fingerprint[] { Junit3Fingerprint, Junit4Fingerprint };
    }
}